### PR TITLE
Add flag to prevent sending URIs

### PR DESCRIPTION
**Pull request by aoking**
Added a flag to prevent URIs (script names) from being sent to New Relic.

The name of the flag setting is `newrelic.collect_script_name` .
The default value is true so as not to break the existing behavior.

https://github.com/newrelic/newrelic-php-agent/issues/190

### CODE REVIEW

**Comment by zsistla** [comment]
Hi @aoking , thanks so much for engaging here and adding your PR.

A couple of process notes:
1) We have internal specs that dictate what config values (in our case INI values) that we can accept.  We are reviewing that document to see if this additional INI value is acceptable in that scope.
2) We do development on the `dev` branch as detailed [here](https://github.com/newrelic/newrelic-php-agent/blob/main/CONTRIBUTING.md#pull-requests).  Can you please set your PR against `dev` and rebase it?
3) Would you be able to add test cases that verify your addition?

Again, thanks so much for contributing!!!

**Comment by zsistla** [comment]
ok jenkins

**Comment by aoking** [comment]
@zsistla 
I re-set it to the dev branch. I also did a rebase.
I did not understand how to write the test. I'm sorry.

**Comment by zsistla** [comment]
Hi @aoking , thanks for updating to `dev`.

For test cases, when adding new functionality, will will typically add cases to one or both of the following:
1) our [integration test suite](https://github.com/newrelic/newrelic-php-agent/tree/main/tests/integration
), for instance a test like [this](https://github.com/newrelic/newrelic-php-agent/blob/main/tests/integration/ini/test_transaction_events_disabled.php) or [this](https://github.com/newrelic/newrelic-php-agent/blob/main/tests/integration/ini/test_transaction_tracer_max_segments.php).
2) unit [tests](https://github.com/newrelic/newrelic-php-agent/tree/main/agent/tests)

On further review of this, it seems that there might already be functionality that you want.

Please correct me if I'm wrong, but your PR seems to attempt to strip all transaction names regardless of whether they contained sensitive data or not.

Configuring `newrelic.transaction_events.enabled = false` in the ini already seems to achieve that effect.
```
; Setting: newrelic.transaction_events.enabled
; Type   : boolean
; Scope  : per-directory
; Default: true
; Info   : Collect and report transaction analytics event data. Event data
;          allows the New Relic UI to show additional information such as
;          histograms. This setting was formerly called
;          newrelic.analytics_events.enabled.
;
;newrelic.transaction_events.enabled = true
```

Alternatively, instead of dropping the transaction name for all functions, you could try calling `newrelic_name_transaction` following the details [here](https://docs.newrelic.com/docs/agents/php-agent/php-agent-api/newrelic_name_transaction/).  You would then be able to rename the functions that would normally generate sensitive data and still track usage and performance of your system.

Please let me know if these would be solutions for your original issue #190

**Comment by aoking** [comment]
I've added unit tests.

File: agent/tests/test_txn.c [hunk 1]
```diff
@@ -161,6 +162,41 @@ static void test_max_segments_config_values(TSRMLS_D) {
   tlib_php_request_end();
 }
 
+
+static void test_set_initial_path(TSRMLS_D) {
+  nrtxn_t* txn;
+  zval* server;  
+  /*
+   * Test : max_segments_cli defaults correctly.
```
Comment (mfulb) [inline agent/tests/test_txn.c:170]: This comment doesn't seem correct for this function - was it copied accidentally?

**Comment by Ak-x** [comment]
Thank you @aoking, for updating the unit test and your overall contribution! 

Our team will be reviewing this issue later this week, and we expect to have this completed next week.

**Comment by Ak-x** [comment]
Thank you @aoking, for updating the unit test and your overall contribution! 

Our team will be reviewing this issue later this week, and we expect to have this completed next week.

File: agent/tests/test_txn.c (hunk 1 above)
Comment (aoking) [reply agent/tests/test_txn.c]: I'm sorry. I fixed it.

**Comment by zsistla** [comment]
ok jenkins

**Comment by zsistla** [comment]
Hi @aoking,

Thanks again for your contribution and your patience as we review it.

Our internal run is required to pass before merging and it detected some valgrind errors in the newly added test function `test_set_initial_path` while running the unit tests.
Here is a snippet:
```
                      test_txn:     16 of     18 tests passed,      2 failed  ./tests/test_txn
                                           with 21 errors from 21 contexts in tests/test_txn.valgrind.log
==1689== Memcheck, a memory error detector
==1689== Copyright (C) 2002-2013, and GNU GPL'd, by Julian Seward et al.
==1689== Using Valgrind-3.10.1 and LibVEX; rerun with -h for copyright info
==1689== Command: ./tests/test_txn
==1689== Parent PID: 1485
==1689== 
==1689== Invalid read of size 8
==1689==    at 0x579600: add_assoc_string_ex (zend_API.c:1291)
==1689==    by 0x4F0879: test_set_initial_path (test_txn.c:182)
==1689==    by 0x4F0A25: test_main (test_txn.c:216)
==1689==    by 0xC848E2: test_main_parallel_driver_helper (tlib_main.c:189)
==1689==    by 0xC84A6A: test_main_parallel_driver (tlib_main.c:236)
==1689==    by 0xC846FC: main (tlib_main.c:142)
==1689==  Address 0x98a6780 is 0 bytes inside a block of size 32 free'd
==1689==    at 0x4A07D30: free (in /opt/nr/camp/lib64/valgrind/vgpreload_memcheck-amd64-linux.so)
==1689==    by 0x53B9B5: _efree (zend_alloc.c:2437)
==1689==    by 0x55CA91: i_zval_ptr_dtor (zend_execute.h:80)
==1689==    by 0x55CA91: _zval_ptr_dtor (zend_execute_API.c:424)
==1689==    by 0x583B15: i_zend_hash_bucket_delete (zend_hash.c:182)
==1689==    by 0x583B15: zend_hash_bucket_delete (zend_hash.c:192)
==1689==    by 0x585601: zend_hash_graceful_reverse_destroy (zend_hash.c:613)
==1689==    by 0x55C28B: shutdown_executor (zend_execute_API.c:244)
==1689==    by 0x572F95: zend_deactivate (zend.c:960)
==1689==    by 0x4F41DA: php_request_shutdown (main.c:1899)
==1689==    by 0xC852F5: tlib_php_request_end_impl (tlib_php.c:405)
==1689==    by 0x4F082A: test_max_segments_config_values (test_txn.c:162)
==1689==    by 0x4F0A20: test_main (test_txn.c:215)
==1689==    by 0xC848E2: test_main_parallel_driver_helper (tlib_main.c:189)
==1689==    by 0xC84A6A: test_main_parallel_driver (tlib_main.c:236)
==1689==    by 0xC846FC: main (tlib_main.c:142)
```

You can see the full output if you run `make valgrind` in your agent build directory.  Please see if you can address the issues.

**Comment by aoking** [comment]
@zsistla 
Sorry, I could not reproduce it in my environment. 
Could you please support me on this?

**Comment by ZNeumann** [comment]
Hi @aoking , thank you for your patience. The above error seems to be occurring with PHP 5.X. If you rebase against the latest dev branch and rebuild with PHP 5.6, are you then able to reproduce the error from `make valgrind`?

**Pull request closed without merging**